modelci/hub/converter.py

```diff
@@ -229,27 +233,31 @@ def from_torch_module(
         save_path.parent.mkdir(parents=True, exist_ok=True)
         save_path_with_ext = save_path.with_suffix('.onnx')
 
-        dummy_tensors, input_names = list(), list()
+        dummy_tensors, input_names, output_names = list(), list(), list()
         for input_ in inputs:
             dtype = model_data_type_to_torch(input_.dtype)
             dummy_tensors.append(torch.rand(batch_size, *input_.shape[1:], requires_grad=True, dtype=dtype))
             input_names.append(input_.name)
-
+        for output_ in outputs:
+            output_names.append(output_.name)
+        if model_input is None:
+            model_input = tuple(dummy_tensors)
         torch.onnx.export(
             model,  # model being run
-            tuple(dummy_tensors),  # model input (or a tuple for multiple inputs)
+            model_input,  # model input (or a tuple for multiple inputs)
             save_path_with_ext,  # where to save the model (can be a file or file-like object)
             export_params=True,  # store the trained parameter weights inside the model file
             opset_version=opset,  # the ONNX version to export the model to
             do_constant_folding=True,  # whether to execute constant folding for optimization
             input_names=input_names,  # the model's input names
-            output_names=['output'],  # the model's output names
+            output_names=output_names,  # the model's output names
             keep_initializers_as_inputs=True,
             **export_kwargs
         )
 
         if optimize:
-            network = ONNXConverter.optim_onnx(save_path_with_ext)
+            onnx_model = onnx.load(str(save_path_with_ext))
```

Comment:
has this been tested？
which onnx version do you use？

Reply:
Yes, it has been tested, this is because `ONNXConverter.optim_onnx` only receives `onnx.ModelProto` object as input instead of PosixPath, It's a fix.

https://github.com/cap-ntu/ML-Model-CI/blob/8165e367ae14659ef27843e8774084b640ef0a76/modelci/hub/converter.py#L307